integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  EntityMap,
+  InterpretationPluginConfig,
+  Jovo,
+  JovoError,
+  NluData,
+  NluPlugin,
+} from '@jovotech/framework';
+
+import {
+  AnalyzeConversationResult,
+  ConversationAnalysisClient,
+  ConversationAnalysisClientOptionalParams,
+  ConversationPrediction,
+  ConversationTaskParameters,
+  ConversationalTask,
+} from '@azure/ai-language-conversations';
+
+import { AzureKeyCredential, KeyCredential, TokenCredential } from '@azure/core-auth';
+import { v4 as uuidV4 } from 'uuid';
+import { INTENT_NONE, PROJECTKIND_CONVERSATION, PARTICIPANT_USER } from './constants';
+
+export interface MicrosoftCluLibraryConfig {
+  taskParameters: ConversationTaskParameters;
+  options?: ConversationAnalysisClientOptionalParams;
+}
+
+export interface MicrosoftCluNluConfig extends InterpretationPluginConfig {
+  endpoint: string;
+  credential: TokenCredential | KeyCredential | string;
+  fallbackLanguage: string;
+  libraryConfig: MicrosoftCluLibraryConfig;
+}
+
+export class MicrosoftCluNlu extends NluPlugin<MicrosoftCluNluConfig> {
+  getDefaultConfig(): MicrosoftCluNluConfig {
+    return {
+      ...super.getDefaultConfig(),
+      fallbackLanguage: 'en',
+      endpoint: '',
+      libraryConfig: {
+        taskParameters: {
+          projectName: '',
+          deploymentName: '',
+        },
+      },
+    };
+  }
+
+  async processText(jovo: Jovo, text: string): Promise<NluData | undefined> {
+    if (!this.config.endpoint) {
+      throw new JovoError({
+        message: `Can not send request to Microsoft-CLU. The endpoint is missing.`,
+      });
+    }
+
+    if (!this.config.libraryConfig.taskParameters.projectName) {
+      throw new JovoError({
+        message: `Can not send request to Microsoft-CLU. The projectName is missing.`,
+      });
+    }
+
+    if (!this.config.libraryConfig.taskParameters.deploymentName) {
+      throw new JovoError({
+        message: `Can not send request to Microsoft-CLU. The deploymentName is missing.`,
+      });
+    }
+
+    const result = await this.sendTextToServer(jovo, text);
+
+    if (result.prediction.projectKind === PROJECTKIND_CONVERSATION) {
```

Comment:
Is there a case where `result` or `result.prediction` could be undefined?

Reply:
@aswetlow No. Both predication and projectKind are required and don't support undefined.

```ts
export declare interface AnalyzeConversationResult {
    /** The conversation utterance given by the caller. */
    query: string;
    /** The system detected language for the query in BCP 47 language representation.. */
    detectedLanguage?: string;
    /** The prediction result of a conversation project. */
    prediction: BasePredictionUnion;
}

/** This is the base class of prediction */
export declare interface BasePrediction {
    /** Polymorphic discriminator, which specifies the different types this object can be */
    projectKind: "Conversation" | "Orchestration";
    /** The intent with the highest score. */
    topIntent?: string;
}

export declare type BasePredictionUnion = ConversationPrediction | OrchestrationPrediction;

/** Represents the prediction section of a Conversation project. */
export declare interface ConversationPrediction extends BasePrediction {
    /** Polymorphic discriminator, which specifies the different types this object can be */
    projectKind: "Conversation";
    /** The intent classification results. */
    intents: ConversationIntent[];
    /** The entity extraction results. */
    entities: ConversationEntity[];
}

/** This represents the prediction result of an Orchestration project. */
export declare interface OrchestrationPrediction extends BasePrediction {
    /** Polymorphic discriminator, which specifies the different types this object can be */
    projectKind: "Orchestration";
    /** A dictionary that contains all intents. A key is an intent name and a value is its confidence score and target type. The top intent's value also contains the actual response from the target project. */
    intents: {
        [propertyName: string]: TargetIntentResultUnion;
    };
}
```